add dbt-hive in vendor supported profiles (#1898)
dbt-labs/docs.getdbt.com: website/docs/reference/warehouse-profiles/hive-profile.md

---
title: "Apache Hive Profile"
id: "hive-profile"
---

## Overview of dbt-hive

**Maintained by:** Cloudera    
**Author:** Cloudera    
**Source:** [Github](https://github.com/cloudera/dbt-hive)    
**dbt Cloud:** Currently un-supported    
**dbt Slack channel** [Link to channel](https://getdbt.slack.com/)     

![dbt-hive stars](https://img.shields.io/github/stars/cloudera/dbt-hive?style=for-the-badge)

## Connection Methods

> dbt-hive can connect to Apache Hive and Cloudera Data Platform clusters.

Reviewer: Can we combine thee sentences?
```suggestion
dbt-hive can connect to Apache Hive and Cloudera Data Platform clusters. The [Impyla](https://github.com/cloudera/impyla/) library is used to establish connections to Hive.
```

> 

Reviewer: ```suggestion
```

> The [Impyla](https://github.com/cloudera/impyla/) library is used to establish connections to Hive.

Reviewer: ```suggestion
```

> Two transport mechanisms are supported:

Reviewer: Let's make this sentence active.
```suggestion
dbt-hive supports two transport mechanisms:
```

- [`ldap`](#ldap) Authentication via LDAP

### Insecure

This method is only recommended if you have a local install of Hive and want to test out the dbt-hive adapter. 

<File name='~/.dbt/profiles.yml'>

```yaml
your_profile_name:
  target: dev
  outputs:
    dev:
      type: hive
      host: localhost
      port: [port]
      schema: [schema name]
      
```

</File>

### LDAP

LDAP allows you to authenticate with a username and password when Hive is [configured with LDAP Auth](https://cwiki.apache.org/confluence/display/Hive/Setting+Up+HiveServer2). LDAP is supported over Binary & HTTP connection mechanisms.

This is the recommended authentication mechanism to use with Cloudera Data Platform (CDP).

<File name='~/.dbt/profiles.yml'>

```yaml
your_profile_name:
  target: dev
  outputs:
    dev:
     type: hive
     host: [host name]
     http_path: [optional, http path to Hive]
     port: [port]
     auth_type: ldap
     use_http_transport: [true / false]
     use_ssl: [true / false] # TLS should always be used with LDAP to ensure secure transmission of credentials
     username: [username]
     password: [password]
     schema: [schema name]
```

</File>

> Note: When creating workload user in CDP ensure that the user has CREATE, SELECT, ALTER, INSERT, UPDATE, DROP, INDEX, READ and WRITE permissions. If the user is required to execute GRANT statements, see for instance (https://docs.getdbt.com/reference/resource-configs/grants) or (https://docs.getdbt.com/reference/project-configs/on-run-start-on-run-end) appropriate GRANT permissions should be configured. When using Apache Ranger, permissions for allowing GRANT are typically set using "Delegate Admin" option. 

Reviewer: ```suggestion
Note: When creating workload user in CDP, make sure the user has CREATE, SELECT, ALTER, INSERT, UPDATE, DROP, INDEX, READ and WRITE permissions. If you need the user to execute GRANT statements, you should also configure the appropriate GRANT permissions for them. When using Apache Ranger, permissions for allowing GRANT are typically set using "Delegate Admin" option. For more information, see [`grants`](/reference/resource-configs/grants) and [on-run-start & on-run-end](/reference/project-configs/on-run-start-on-run-en).
```

---
title: "Apache Hive Profile"
id: "hive-profile"
---

## Overview of dbt-hive

**Maintained by:** Cloudera    
**Author:** Cloudera    
**Source:** [Github](https://github.com/cloudera/dbt-hive)    
**dbt Cloud:** Currently un-supported    
**dbt Slack channel** [Link to channel](https://getdbt.slack.com/)     

![dbt-hive stars](https://img.shields.io/github/stars/cloudera/dbt-hive?style=for-the-badge)

## Connection Methods

dbt-hive can connect to Apache Hive and Cloudera Data Platform clusters.

The [Impyla](https://github.com/cloudera/impyla/) library is used to establish connections to Hive.

Two transport mechanisms are supported:
- binary
- HTTP(S)

The default mechanism is `binary`. To use HTTP transport, use the boolean option `use_http_transport: [true / false]`.

## Authentication Methods

> dbt-hive supports three authentication mechanisms:

Reviewer 2: We have only mentioned 2 authentication methods,

Author: Yes, we haven't implemented kerberos yet, fixed the number
